numpy/lib/utils.py

```diff
@@ -193,7 +193,44 @@ def deprecate(*args, **kwargs):
     else:
         return _Deprecate(*args, **kwargs)
 
-deprecate_with_doc = lambda msg: _Deprecate(message=msg)
+def deprecate_with_doc(msg):
+    """
+    Returns new _Deprecate class object. 
+    The object can be used to Issue a DeprecationWarning, by passing `func`
+    as arguement,this adds warning to `old_name`'s docstring, rebinds 
```

Comment:
```suggestion
    as argument, this adds warning to `old_name`'s docstring, rebinds 
```
Minor typo fix.
Secondly, what exactly is this `old_name` parameter referenced here?

Reply:
old_name is an instance variable for objects (here the object returned by `deprecate_with_doc`)  of _Deprecate class.

Reply:
It seems to be a parameter of `np.deprecate` ([docs](https://numpy.org/devdocs/reference/generated/numpy.deprecate.html)) as well, but not for `deprecate_with_doc`.
Maybe something along the lines of:
```suggestion
    as argument, this adds warning to the to-be decorated function's docstring, rebinds 
```